### PR TITLE
Fixes #27764, #27798, #27391 - host form update on change

### PR DESCRIPTION
**Pull request by ezr-ondrej**
We need to select host form more reliably as there are cases where some components are adding subforms.

### CODE REVIEW

**Comment by tbrisker** [comment]
Nested forms are invalid HTML. This sounds like it should be changed where we add forms inside the form.

**Comment by ezr-ondrej** [comment]
> Nested forms are invalid HTML. This sounds like it should be changed where we add forms inside the form.

I agree, but that is going to take some time as it is pattern-fly pagination component.

File: app/views/hosts/_form.html.erb [hunk 1]
```diff
@@ -4,7 +4,7 @@
 <%= render "hosts/progress" %>
 
 <% Taxonomy.as_taxonomy @organization , @location do %>
-
+<div id="host_form">
   <%= form_for @host, :url => (@host.new_record? ? hosts_path : host_path(:id => @host.id)), :html => {:data => {:id => @host.try(:id), :type_changed => !!@host.type_changed?, :submit => 'progress_bar'}} do |f| %>
```
Comment (tbrisker) [inline app/views/hosts/_form.html.erb]: I think it would be better to add an id to the form instead of wrapping it with another div that may affect layout

**Review by tbrisker — CHANGES_REQUESTED** [review]
Can you open an issue to PF to address the issue for the long term? I think there other places where we select the host form that may need updates.

**Comment by tbrisker** [comment]
Looks like this also fixes [#27764](https://projects.theforeman.org/issues/27764), please update the commit message to to `Fixed #27764, #27798 ...` so it gets attached and closed.

File: app/views/hosts/_form.html.erb (hunk 1 above)
Comment (tbrisker) [inline app/views/hosts/_form.html.erb]:
```suggestion
  <%= form_for @host, :url => (@host.new_record? ? hosts_path : host_path(:id => @host.id)), :html => {:id => 'host_form', :data => {:id => @host.try(:id), :type_changed => !!@host.type_changed?, :submit => 'progress_bar'}} do |f| %>
```

Comment (ezr-ondrej) [reply app/views/hosts/_form.html.erb]: You want only one space? :eyes:

Comment (tbrisker) [reply app/views/hosts/_form.html.erb]: no, i want the id on the form instead of a wrapper div :)

Comment (ezr-ondrej) [reply app/views/hosts/_form.html.erb]: ah, it was too far xD I didn't see.
What about this?

**Review by tbrisker — COMMENTED** [review]
please also update the selectors in other functions using the `form` selector. Also, this will break the hostgroup form which uses the same js code - so we need to use a shared class on both forms.

**Comment by ezr-ondrej** [comment]
I've made the shared class, updated other occurences.

**Review by tbrisker — APPROVED** [review]
Thanks @ezr-ondrej ! tested and seems to fix all the issues.

**Comment by ezr-ondrej** [comment]
:green_heart:

**Comment by tbrisker** [comment]
1.23 - 17b48be66

**Comment by ezr-ondrej** [comment]
@tbrisker the issue seems to be in 1.22 already according to [discourse thread](https://community.theforeman.org/t/create-host-with-host-groups-and-compute-profile-vcenter-storage-tab-empty/14752/9), should we backport it to 1.22 as well?

**Comment by tbrisker** [comment]
@ezr-ondrej can you open a cherry-pick pr to 1.22-stable? this did not cleanly apply to 1.23, and i expect 1.22 would be even more difficult.

**Pull request merged**